Sources/SwiftRepo/Repository/Store/PersistentStore.swift

```diff
@@ -1,189 +1,124 @@
-        case let value?:
-            guard let wrapped = value.wrapped else { return nil }
-            try secondLevelStore?.set(key: key, value: wrapped)
-            try save(wrapped: wrapped, url: url(for: key))
-            return value
-        case .none:
-            try secondLevelStore?.set(key: key, value: nil)
-            let url = url(for: key)
-            try FileManager.default.removeItem(atPath: url.path)
-            return nil
-        }
+    public func get(key: Key) throws -> Value? {
+        let keyData = try encoder.encode(key)
+        guard let valueData = try modelContext.fetch(
+            FetchDescriptor(predicate: TimestampedValue.predicate(forKeyData: keyData))
+        ).first else { return nil }
+        return try decoder.decode(Value.self, from: valueData.value)
     }
-
+    
+    @discardableResult
     @MainActor
-    public func get(key: Key) throws -> Value? {
-        if let wrapped = try secondLevelStore?.get(key: key) { return PersistentValue(initial: .wrapped(wrapped)) }
-        let url = url(for: key)
-        switch FileManager.default.fileExists(atPath: url.path) {
-        case true: return load(url: url)
-        case false: return nil
+    public func set(key: Key, value: Value?) throws -> Value? {
+        if let value {
+            try modelContext.insert(TimestampedValue(key: key, value: value, encoder: encoder))
+            try modelContext.save()
+        } else {
+            let keyData = try encoder.encode(key)
+            try evict(for: keyData)
         }
+        return value
     }
-
+    
     @MainActor
     public func age(of key: Key) throws -> TimeInterval? {
-        let url = url(for: key)
-        let attr = try FileManager.default.attributesOfItem(atPath: url.path)
-        guard let date = attr[FileAttributeKey.modificationDate] as? Date else { return nil }
-        return Date().timeIntervalSince(date)
+        let keyData = try encoder.encode(key)
+        let result = try modelContext.fetch(FetchDescriptor(predicate: TimestampedValue.predicate(forKeyData: keyData)))
+        guard let result = result.first else { return nil }
+        return Date.now.timeIntervalSince(result.timestamp)
     }
-
-    public func clear() async throws {
-        try await secondLevelStore?.clear()
-        try FileManager.default.removeItem(atPath: location.directoryURL.path)
-        fatalError("TODO delete the directory")
-    }
-
+    
     @MainActor
-    public var keys: [Key] {
-        (try? FileManager.default.contentsOfDirectory(atPath: location.directoryURL.path)) ?? []
+    public func clear() async throws {
+        try modelContext.delete(model: TimestampedValue.self)
+        try modelContext.save()
     }
-
-    // MARK: - File management
-
-    @MainActor
-    private func url(for key: Key) -> URL {
-        let url = location.directoryURL
-        return url.appendingPathComponent(key)
+    
+    // MARK: - Constants
+    
+    @Model
+    class TimestampedValue {
```

Comment:
Making it a top-level type would mean it needs to be generic, because it inherits the `Key` and `Value` types from the `PersistentStore`. I'm wondering if making a generic SwiftData model is kind of an anti-pattern

Reply:
Hmm. Yeah, I dunno. I'm good with leaving this as is.